content/docs/reference/design-docs/train-sim-v3/driver-behavior-modules.en.md

```diff
@@ -3,36 +3,82 @@ title: "Driver behavior modules"
-end
+Driver behavior modules are responsible for making driving decisions. 
+It's main responsibility, given the state of the train and an instruction, is to react to the instruction.
+This reaction is expressed as a new train state.
 
-DriverBehaviorModule[driver behavior module]
-Decision[decision]
+To perform this critical task, it needs access to additional context:
 
-BuilderInputs --> DriverBehaviorModule
-CallInputs --> DriverBehaviorModule --> Decision
+- the physical properties of the path, which are used to make coasting decisions, and to model natural forces.
+- a slowdown coefficient, which is used to adjust how much the train is slowed down compared to a 
+full power simulation.
 
-```
+The driver behavior modules are supposed to have different implementations, which would interpret the slow down coefficient differently.
 
-Driver behavior modules are responsible for making driving decisions. It's main responsibility, given the state of the train,
-is to provide a driving instruction (a braking or traction force). To perform this critical task, it needs access to additional context:
 
-- the physical properties of the path, which are used to make coasting decisions
-- driving instructions
+### API
 
-## Design decisions
+One driver behavior module is instantiated per driving instruction.
+It takes at initialization:
+- a slowdown coefficient
+- the driving instruction
+- the path properties
+
+It has two public methods:
+- `enact_decision(current_state: TrainState, t: float) -> (TrainState, float)`
+
+  Which returns what the next train state would be if there was only this one instruction to follow, and the time delta
+  to reach this state.
+
+- `truncate_integration_step(current_state: TrainState, potential_state: TrainState, t: float, dt: float) -> (TrainState, float)`
+
+  Which returns a state and time delta which respects the instruction, and is as close as possible to the potential state. 
+
+
+### Loop
+At a given train state, we know which driving instructions are enforced.
+
+For each enforced driving instruction, we query the corresponding driver behavior module.
+
+This gives a set of different train states. From this, we coalesce a single train state which respects all instructions.
+
+To do so, we:
+1. Find the states which are most constraining for "constraining properties" (speed and pantograph state).
+  - Most constraining state regarding speed is the one with the lowest acceleration (taking sign into account).
+  - Most constraining state regarding pantograph state is the one which sets the pantograph down the earliest.
+2. Interpolate the **constraining** states to the smallest `dt` they are associated with.
+3. Merge the **constraining** states into a single state: 
+  - for speed, we take the lowest acceleration
+  - for pantograph state, we take the earliest pantograph state
+  - other properties should be identical
+4. Submit this state for truncation to all driver behavior modules. 
```

Comment:
One by one, by chaining them, feeding the output of `truncate_integration_step()` to the next DBM?